apps/dsb-message-broker/src/utils/topic.schema.service.ts

```diff
@@ -1,53 +1,137 @@
-import { Injectable } from '@nestjs/common';
-import { ModuleRef } from '@nestjs/core';
-import Ajv, { JSONSchemaType, AnySchema, ValidateFunction } from 'ajv';
+import path from 'path';
+import fs from 'fs';
+import { BadRequestException, Injectable } from '@nestjs/common';
+import Ajv, { JSONSchemaType, ValidateFunction } from 'ajv';
 import addFormats from 'ajv-formats';
+import Libxml from 'node-libxml';
 
 import { AddressBookService } from '../addressbook/addressbook.service';
 
+const libxml = new Libxml();
+
+class JSONValidator {
+    constructor(public readonly validate: ValidateFunction<unknown>) {}
+}
+class XMLValidator {
+    constructor(public readonly schema: string, public readonly pathToSchema: string) {}
+}
+
 @Injectable()
 export class TopicSchemaService {
     private readonly ajv = new Ajv({
         multipleOfPrecision: 1000000000
     });
-    private readonly _validators = new Map<string, Record<string, ValidateFunction<unknown>>>();
+    private readonly _validators = new Map<string, Record<string, JSONValidator | XMLValidator>>();
 
-    constructor(
-        private readonly moduleRef: ModuleRef,
-        private readonly addressbook: AddressBookService
-    ) {
+    constructor(private readonly addressbook: AddressBookService) {
         addFormats(this.ajv, { mode: 'fast', formats: ['date', 'time'], keywords: true });
+
+        const xsdDirectory = path.resolve(__dirname, `../../../xsd_files`);
```

Comment:
Don not use relative path like this, since this could be invalid one for e.g when running in Docker image. Use `os.tmpdir()` instead to get the temporary fs path.

Reply:
fixed